server/index.js

```diff
@@ -48,13 +48,119 @@ app.get('/api/products/:productId', (req, res, next) =>{
   db.query(sql, values)
     .then( result => {
       if(result.rows.length<1){
-        return next(new ClientError(`cannot find with provided input`, 404))
+        return next(new ClientError(`Cannot find with provided input`, 404))
       }
       res.status(200).json(result.rows[0])
     })
     .catch(err => next(err))
 });
 
+app.get('/api/cart', (req, res, next) => {
+  const sql = `
+           select "c"."cartItemId",
+                  "c"."price",
+                  "p"."productId",
+                  "p"."image",
+                  "p"."name",
+                  "p"."shortDescription"
+             from "cartItems" as "c"
+             join "products" as "p" using ("productId")
+            where "c"."cartId" = $1
+          `
+  const values = [req.session.cartId]
+  if(!req.session.cartId){
+    return res.json([])
+  }
+
+  db.query(sql, values)
+    .then ( result => {
+
+      return res.status(200).json(result.rows)
+    })
+    .catch(err => next(err))
+})
+
+app.post('/api/cart', (req, res, next) => {
+  const productId = req.body.productId
+  const values = [productId]
+  const sql = `
+            SELECT "price"
+            FROM "products"
+            WHERE "productId" = $1
+            `
+
+  if(productId<1){
+    return res.status(400).json({ error: "Id must be a positive number"})
+  }
+
+  db.query(sql, values)
+    .then( result => {
+      if(result.rows.length<1){
+        return next(new ClientError(`Cannot find product with that provided input`, 400))
+      }
+      if(req.session.cartId){
+        return {
+          cartId: req.session.cartId,
+          price: result.rows[0].price
+        }
+      }
+      const createSql = `
+                  insert into "carts" ("cartId", "createdAt")
+                  values (default, default)
+                  returning "cartId"
+                  `
+
+      return (db.query(createSql)
+        .then(cartResult =>{
+          return {
+            cartId: cartResult.rows[0].cartId,
+            price: result.rows[0].price
+          }
+        })
+        .catch(error => console.error(error))
```

Comment:
You can leave out this `catch()`. The last one in your promise chain will get the error.